plugins/package-managers/gradle-inspector/src/main/kotlin/GradleInspector.kt

```diff
@@ -0,0 +1,250 @@
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.gradleinspector
+
+import OrtDependency
+import OrtDependencyTreeModel
+
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.util.SortedSet
+import java.util.concurrent.TimeUnit
+
+import org.apache.logging.log4j.kotlin.Logging
+
+import org.gradle.tooling.GradleConnector
+import org.gradle.tooling.events.ProgressListener
+import org.gradle.tooling.internal.consumer.DefaultGradleConnector
+
+import org.ossreviewtoolkit.analyzer.AbstractPackageManagerFactory
+import org.ossreviewtoolkit.analyzer.PackageManager
+import org.ossreviewtoolkit.downloader.VersionControlSystem
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Issue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.Severity
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.PackageManagerConfiguration
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.utils.ort.createOrtTempFile
+
+/**
+ * The names of Gradle (Groovy, Kotlin script) build files for a Gradle project.
+ */
+private val GRADLE_BUILD_FILES = listOf("build.gradle", "build.gradle.kts")
+
+/**
+ * The names of Gradle (Groovy, Kotlin script) settings files for a Gradle build.
+ */
+private val GRADLE_SETTINGS_FILES = listOf("settings.gradle", "settings.gradle.kts")
+
+/**
+ * The name of the option to specify the Gradle version.
+ */
+const val OPTION_GRADLE_VERSION = "gradleVersion"
+
+/**
+ * The [Gradle](https://gradle.org/) package manager for Java.
+ *
+ * This package manager supports the following [options][PackageManagerConfiguration.options]:
+ * - *gradleVersion*: The version of Gradle to use when analyzing projects. Defaults to the version defined in the
+ *   Gradle wrapper properties.
+ */
+class GradleInspector(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig) {
+    private companion object : Logging
+
+    class Factory : AbstractPackageManagerFactory<GradleInspector>("GradleInspector", isEnabledByDefault = false) {
+        // Gradle prefers Groovy ".gradle" files over Kotlin ".gradle.kts" files, but "build" files have to come before
+        // "settings" files as we should consider "settings" files only if the same directory does not also contain a
+        // "build" file.
+        override val globsForDefinitionFiles = GRADLE_BUILD_FILES + GRADLE_SETTINGS_FILES
```

Comment:
On the first glance I didn't get what the comment for `globsForDefinitionFiles` is trying to say. I believe it could become more obvious by droping `GRADLE_BUILD_FILES` and `GRADLE_SETTINGS_FILES` and have the comment next to one list of all globs. Either
`val GLOBS_FOR_DEFINITON_FILES` as a file level constant, or directly inline. What do you think?

Reply:
Note that this comment was copied as-is from `Gradle.kt`. And at least over there, `GRADLE_SETTINGS_FILES` is also used separately. Thus I'd prefer to keep the constants separate and aligned with `Gradle.kt`.

Reply:
> Thus I'd prefer to keep the constants separate and aligned with Gradle.kt.

Ok, I was reviewing it under the assumption that `Gradle.kt` is gonna be deleted anyway.
If it makes sense for the time being I'm fine with that.